Metricbeat test failures in travis job "metricbeat integration-tests-environment coverage-report"

Test killed with quit: ran too long (10m0s).

````
ok  	github.com/elastic/beats/metricbeat/module/vsphere/virtualmachine	0.024s	coverage: 44.4% of statements
?   	github.com/elastic/beats/metricbeat/module/windows	[no test files]
?   	github.com/elastic/beats/metricbeat/module/windows/perfmon	[no test files]
?   	github.com/elastic/beats/metricbeat/module/windows/service	[no test files]
?   	github.com/elastic/beats/metricbeat/module/zookeeper	[no test files]
ok  	github.com/elastic/beats/metricbeat/module/zookeeper/mntr	4.853s	coverage: 84.0% of statements
ok  	github.com/elastic/beats/metricbeat/module/zookeeper/server	1.040s	coverage: 71.9% of statements
?   	github.com/elastic/beats/metricbeat/processor/add_kubernetes_metadata	[no test files]
?   	github.com/elastic/beats/metricbeat/scripts/assets	[no test files]
command [go test -tags integration -cover -coverprofile /tmp/gotestcover-299454606 github.com/elastic/beats/metricbeat/helper]: exit status 1
2019/03/20 12:25:33 httptest.Server blocked in Close after 5 seconds, waiting for connections:
  *net.TCPConn 0xc42000e100 127.0.0.1:42756 in state active
SIGQUIT: quit
PC=0x45c5d1 m=0 sigcode=0
goroutine 0 [idle]:
runtime.futex(0xa82848, 0x0, 0x0, 0x0, 0xc400000000, 0x7f9500000000, 0x0, 0x0, 0x7ffd4505bfd8, 0x410c2b, ...)
	/usr/local/go/src/runtime/sys_linux_amd64.s:530 +0x21
runtime.futexsleep(0xa82848, 0x7f9500000000, 0xffffffffffffffff)
	/usr/local/go/src/runtime/os_linux.go:45 +0x4b
runtime.notesleep(0xa82848)
	/usr/local/go/src/runtime/lock_futex.go:151 +0x9b
runtime.stopm()
	/usr/local/go/src/runtime/proc.go:1952 +0xe5
runtime.findrunnable(0xc420030500, 0x0)
	/usr/local/go/src/runtime/proc.go:2415 +0x50c
runtime.schedule()
	/usr/local/go/src/runtime/proc.go:2541 +0x13b
runtime.park_m(0xc420001c80)
	/usr/local/go/src/runtime/proc.go:2604 +0xb6
runtime.mcall(0x0)
	/usr/local/go/src/runtime/asm_amd64.s:351 +0x5b
goroutine 1 [chan receive]:
testing.(*T).Run(0xc42012c4b0, 0x83d102, 0xb, 0x854390, 0x479701)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc42012c000)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc42012c000, 0xc420063d90)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc420104760, 0xa7b660, 0x4, 0x4, 0x411c49)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc420120080, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
main.main()
	_testmain.go:96 +0x208
goroutine 9 [semacquire]:
sync.runtime_Semacquire(0xc42011de04)
	/usr/local/go/src/runtime/sema.go:56 +0x39
sync.(*WaitGroup).Wait(0xc42011ddf8)
	/usr/local/go/src/sync/waitgroup.go:129 +0x72
net/http/httptest.(*Server).Close(0xc42011ddc0)
	/usr/local/go/src/net/http/httptest/server.go:221 +0x208
github.com/elastic/beats/metricbeat/helper.TestTimeout(0xc42012c4b0)
	/go/src/github.com/elastic/beats/metricbeat/helper/http_test.go:98 +0x294
testing.tRunner(0xc42012c4b0, 0x854390)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0
goroutine 19 [chan receive]:
github.com/elastic/beats/metricbeat/helper.TestTimeout.func1(0x888b80, 0xc420138000, 0xc4200c2700)
	/go/src/github.com/elastic/beats/metricbeat/helper/http_test.go:83 +0x33
net/http.HandlerFunc.ServeHTTP(0xc42002d280, 0x888b80, 0xc420138000, 0xc4200c2700)
	/usr/local/go/src/net/http/server.go:1947 +0x44
net/http.serverHandler.ServeHTTP(0xc420087ad0, 0x888b80, 0xc420138000, 0xc4200c2700)
	/usr/local/go/src/net/http/server.go:2694 +0xbc
net/http.(*conn).serve(0xc4200a55e0, 0x888ec0, 0xc42006ecc0)
	/usr/local/go/src/net/http/server.go:1830 +0x651
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:2795 +0x27b
rax    0xca
rbx    0xa82700
rcx    0x45c5d3
rdx    0x0
rdi    0xa82848
rsi    0x0
rbp    0x7ffd4505bfa0
rsp    0x7ffd4505bf58
r8     0x0
r9     0x0
r10    0x0
r11    0x286
r12    0x0
r13    0xf1
r14    0x11
r15    0x0
rip    0x45c5d1
rflags 0x286
cs     0x33
fs     0x0
gs     0x0
*** Test killed with quit: ran too long (10m0s).
FAIL	github.com/elastic/beats/metricbeat/helper	600.004s
test failed
../libbeat/scripts/Makefile:198: recipe for target 'integration-tests' failed
make: *** [integration-tests] Error 1
make: *** [integration-tests-environment] Error 2
make: Leaving directory `/home/travis/gopath/src/github.com/elastic/beats/metricbeat'
The command "make $TARGETS" exited with 2.

```

--- a/metricbeat/helper/http_test.go
+++ b/metricbeat/helper/http_test.go
@@ -77,10 +77,11 @@ func TestGetAuthHeaderFromTokenNoFile(t *testing.T) {
 }
 
 func TestTimeout(t *testing.T) {
-	c := make(chan struct{})
-	defer close(c)
 	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		<-c
+		select {
+		case <-time.After(5 * time.Nanosecond):
+		case <-r.Context().Done():
+		}
 	}))
 	defer ts.Close()
 
@@ -118,8 +119,11 @@ func checkTimeout(t *testing.T, h *HTTP) {
 
 	done := make(chan struct{})
 	go func() {
-		_, err := h.FetchResponse()
+		response, err := h.FetchResponse()
 		assert.Error(t, err)
+		if response != nil {
+			response.Body.Close()
+		}
 		done <- struct{}{}
 	}()
 

